test/e2e/node_ip_migration.go

```diff
@@ -0,0 +1,680 @@
+								if err != nil {
+									framework.Logf("Poking all pod IPs failed: %q", err)
+								}
+								return err == nil
+							}, pollingTimeout, pollingInterval).Should(BeTrue())
+						})
+				}
+			})
+
+			Context("when EgressIPs are configured", func() {
+				BeforeEach(func() {
+					By("Adding the \"k8s.ovn.org/egress-assignable\" label to the first node")
+					framework.AddOrUpdateLabelOnNode(f.ClientSet, workerNode.Name, "k8s.ovn.org/egress-assignable", "")
+
+					By("Creating an EgressIP object with one egress IP defined")
+					// Assign the egress IP without conflicting with any node IP,
+					// the kind subnet is /16 or /64 so the following should be fine.
+					// pick something at the end of the range to avoid conflicts with the kind / docker network setup.
+					// Exclude current node IPs and the migrationWorkerNodeIP (if already selected).
+					var err error
+					egressIP, err = findLastFreeSubnetIP(
+						externalContainerName,
+						externalContainerIPs[ipAddrFamily],
+						[]string{workerNodeIPs[ipAddrFamily], secondaryWorkerNodeIPs[ipAddrFamily], migrationWorkerNodeIP},
+					)
+					Expect(err).NotTo(HaveOccurred())
+					framework.Logf("EgressIP will be %s", egressIP)
+
+					var egressIPConfig = fmt.Sprintf(egressIPYamlTemplate, podLabels["app"], egressIP, "app",
+						podLabels["app"], f.Namespace.Name)
+					if err := ioutil.WriteFile(
+						tmpDirIPMigration+"/"+egressIPYaml, []byte(egressIPConfig), 0644); err != nil {
+						framework.Failf("Unable to write CRD config to disk: %v", err)
+					}
+					_, err = framework.RunKubectl("default", "apply", "-f", tmpDirIPMigration+"/"+egressIPYaml)
+					Expect(err).NotTo(HaveOccurred())
+
+					By("Verifying egress IP works")
+					expectedAnswer := "^" + egressIP + ":.*"
+					if utilnet.IsIPv6String(egressIP) {
+						expectedAnswer = "^\\[" + egressIP + "\\]:.*"
+					}
+					Eventually(func() bool {
+						By("Checking the egress IP")
+						res, err := targetExternalContainerConnectToEndpoint(externalContainerName,
+							externalContainerIPs[ipAddrFamily], externalContainerPort, externalContainerEndpoint,
+							podWorkerNode.Name, f.Namespace.Name, expectedAnswer)
+						if err != nil {
+							framework.Logf("Current verification failed with %s", err)
+							return false
+						}
+						return res
+					}, pollingTimeout, pollingInterval).Should(BeTrue())
+				})
+
+				JustAfterEach(func() {
+					By("Deleting the gressip service")
+					framework.RunKubectl("default", "delete", "eip", podLabels["app"])
+
+					By(fmt.Sprintf("Removing the egress assignable label from node %s", workerNode.Name))
+					framework.RunKubectl("default", "label", "node", workerNode.Name, "k8s.ovn.org/egress-assignable-")
+				})
+
+				for _, updateKubeletFirst := range []bool{true, false} {
+					updateKubeletFirst := updateKubeletFirst // Required to avoid race conditions due to pointer assignment.
```

Comment:
not seeing why this is needed

Reply:
@andreaskaris  could you help on Tim's question?